crates/services/consensus_module/poa/src/sync.rs

```diff
@@ -0,0 +1,245 @@
+use std::{
+    sync::Arc,
+    time::Duration,
+};
+
+use fuel_core_types::{
+    fuel_types::BlockHeight,
+    services::block_importer::ImportResult,
+};
+use tokio::{
+    sync::{
+        broadcast,
+        watch,
+    },
+    time::Instant,
+};
+
+use crate::ports::P2pPort;
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum SyncState {
+    NotSynced,
+    Synced,
+}
+
+pub struct SyncTask<T> {
+    min_connected_reserved_peers: usize,
+    time_until_synced: Duration,
+    time_left_until_synced: Duration,
```

Comment:
You can use a timer instead and `reset` it when you need it, like we already do for the `PoA`.

Reply:
Yeah that makes it much cleaner